File: IOT_Protocols/pub-sub/subscriber.js
const mqtt = require('mqtt')
const client = mqtt.connect('mqtt://localhost:1883')
const TOPIC = "diot"
client.on('connect', function () 
{
	//subscribe only after successful connection with broker
	client.subscribe(TOPIC, function (err)
	{
		if(!err)
		{
			console.log(`Successfully subscribed to ${TOPIC}`);
		}
	})
})

client.on('message', function(topic, message)
{
	//message is buffer
	console.log(message.toString())
	client.end()
})

File: IOT_Protocols/study material/Codebase/python/subscriber.py
# Python 3 : Program to use MQTT over Websockets
# Date: 14 Jun 2022
# Description: Mosquitto Broker is tested with websockets connection. Broker is running at 8080
import paho.mqtt.client as mqtt

print("/*-------------------------------------")
print("  MQTT over Websockets: Subscriber client")
print("*--------------------------------------*/")

TOPIC = "diot/fc/df/+/class01"

# Method for handling Subscriber messages
def on_message(client, userdata,message):
 print("Topic: " + message.topic +" || Message: "+ message.payload.decode('utf-8'))

# Method to check if Broker is connected
def on_connect(client, userdata, flags, rc):
 if rc ==0:
  print("connected to the Broker!")
  client.subscribe(TOPIC)
  client.publish("diot/fc/df/gh/class01","Hello")

client = mqtt.Client(transport="websockets")

# Setting the message handlers
client.on_connect = on_connect
client.on_message = on_message

# Request Broker for connection on Port 8080
client.connect("192.168.77.45", 8080)

# To maintain the connection loop
client.loop_forever()
